### ISSUE TITLE
The declaration file can be pruned further

### ISSUE THREAD

**Issue by kran6a** [algora-io/xtrpc#1]
The following endpoint fields can be stripped without affecting type inference:

- `transformer`
- `_input_out` (result of transforming the raw input via the zod schema `.transform()` method)
- `_output_in` (what is returned by the procedure **before** being transformed by the zod output schema)
- `middleware`
- `_ctx_out`
- `_meta`
- `_config` (not sure if it affects type inferency in any way)

Some fields are leaking implementation data if present in the declaration file on top of not being required for inference

**Comment by zcesur** [comment]
Yeah makes sense! Tho we're already removing `middleware` so there's currently no implementation data present I believe?
